pyanaconda/modules/timezone/timezone.py

```diff
@@ -259,3 +277,35 @@ def geolocation_result(self):
         :return GeolocationData: result of the lookup, empty if not ready yet
         """
         return self._geoloc_result
+
+    def get_system_date_time(self):
+        """Get system time as a ISO 8601 formatted string.
+
+        :return: system time as ISO 8601 formatted string
+        :rtype: str
+        """
+        # FIXME: check if this is correct to do ?
+        if self._timezone is None:
+            timezone = "America/New_York"
+        else:
+            timezone = self._timezone
+        # convert to the expected tzinfo format via get_timezone()
+        return datetime.datetime.now(get_timezone(timezone)).isoformat()
```

Comment:
If the timezone is invalid, this will cause an error. Maybe the timezone setter should check the validity of the new timezone and reject it if it is not valid to make sure that the these calls always succeed. @VladimirSlavik, what do you think?

Reply:
Setter rejecting invalid values is a possibility, and probably even a good idea. However, there's more to consider.

Timezone is taken verbatim from kickstart, so there's always a way to sneak in something invalid. That's why I didn't make the setter check before, IIRC. I wonder if fallback-ifying invalid timezone in kickstart would be an option. Then the check would be somewhere in `TimezoneService.set_timezone_with_priority()` and apply to everything.

Theoretically, the install env should have the same data as the payload, so timezone validity should be identical. Additionally, this is slow-moving data defined by a standard, and the actual changes are mostly in the time rules, not zone names. So I think that approach might be fine.

I guess this would be more of a Steve B. topic, but anyway, any thoughts @jstodola?

(Enforcing timezone validity in kickstart should have a release note though, so that we don't forget about it.)

Reply:
Ah, I just realized something. The only way to set time was on time spoke. So this probably worked such that the kickstart data was taken as-is, and then if the spoke was actually entered, the bad zone was fixed on entry so there was no problem setting the time.

(Past tense because the changes around welcome screen, geolocation etc. might have removed this.)